contracts/modules/staking/PooledStaking.sol

```diff
@@ -737,47 +957,519 @@ contract PooledStaking is IPooledStaking, MasterAware {
+      stakerAddress == 0xE1Ad30971b83c17E2A24c0334CB45f808AbEBc87 || // 2.5%
+      stakerAddress == 0x5FAdEA9d64FFbe0b8A6799B8f0c72250F92E2B1d || // 1.7%
+      stakerAddress == 0x9c657DB2B697846BE13Ca0B2bB5a6D17f860a395 || // 1.5%
+      stakerAddress == 0xF99b3a13d46A04735BF3828eB3030cfED5Ea0087 || // 1.4%
+      stakerAddress == 0x8C878B8f805472C0b70eD66a71c0B33da3d233c8 || // 1.4%
+      stakerAddress == 0x4544e2Fae244eA4Ca20d075bb760561Ce5990DC3, // 0.7%
       "You are not authorized to migrate this staker"
     );
 
+    require(stakers[stakerAddress].deposit > 0, "Address has no migratable stake");
+    console.log("Migrating staker %s", stakerAddress);
 
     uint contractsCount = stakers[stakerAddress].contracts.length;
     uint deposit = stakers[stakerAddress].deposit;
 
-    uint[] memory weights = new uint[](contractsCount);
     uint[] memory products = new uint[](contractsCount);
+    uint[] memory stakes = new uint[](contractsCount);
+    uint migratableCount = 0;
+
     for (uint i = 0; i < contractsCount; i++) {
       address oldProductId = stakers[stakerAddress].contracts[i];
-      uint productId = productsV1.getNewProductId(oldProductId);
+      uint productId;
+      try productsV1.getNewProductId(oldProductId) returns (uint v) {
+        productId = v;
+      } catch {
+        console.log("oldProductId %s not found in productsV1, likely deprecated", oldProductId);
+        continue;
+      }
       products[i] = productId;
-      uint stake = stakers[stakerAddress].stakes[oldProductId];
-      weights[i] = stake * 1e18 / deposit;
+      stakes[i] = stakers[stakerAddress].stakes[oldProductId];
+      migratableCount++;
+    }
+
+    ProductInitializationParams[] memory params = new ProductInitializationParams[](migratableCount);
+    uint migrateAtIndex = 0;
+
+    for (uint i = 0; i < contractsCount; i++) {
+      if (stakes[i] == 0) {
+        continue;
+      }
+      uint price = getV1ProceForProduct(products[i]);
+      params[migrateAtIndex] = ProductInitializationParams(
+        products[i], // productId
+        uint8(min(stakes[i] * 1e18 / deposit / 1e16, 100)), // weight (0-100)
+        price, // initialPrice
+        price // targetPrice
+      );
+      migrateAtIndex++;
     }
 
-    // IStakingPool stakingPool = IStakingPool(cover.createStakingPool(stakerAddress));
+    {
+      IStakingPool stakingPool = IStakingPool(cover.createStakingPool(stakerAddress, params));
+      uint stakePositionNFTId;
+
+      {
+        // [todo] This might require an offset to start the first group at the time of v2 deploy
+        uint GROUP_SIZE = 91 days;
+        uint firstActiveGroupId = block.timestamp / GROUP_SIZE;
+
+        // Use the groupId provided as a paramtere if the user is migrating to v2 himself
+        // Use next id after the first active gorup id for those in the initial migration list
+        uint groupIdInEffect = stakerAddress == msg.sender ? groupId : firstActiveGroupId + 1;
```

Comment:
so for addresses that are not in the initial migration list, the groupId is self-defined (parameter).

Is stakingPool.deposit supposed to revert if that groupId is wrong (not in the right range)?

Reply:
I think it should. At the moment it looks like it only checks whether the group is in the future.

`    require(groupId >= firstActiveGroupId, "StakingPool: Requested group has expired");`

Added a todo in StakingPool.sol.